Source/cursor.cpp

```diff
@@ -178,7 +177,7 @@ void ResetCursor()
 void NewCursor(int cursId)
 {
 	pcurs = cursId;
-	cursSize = GetInvItemSize(cursId);
+	cursSize = cursId == CURSOR_NONE ? Size { 0, 0 } : GetInvItemSize(cursId);

```

Comment:
Given the check is fairly simple and the function currently returns both inventory item sizes and cursor sizes you could move the `cursId == CURSOR_NONE` check inside `GetInvItemSize` to save repeating it here. Most of the callers of that function are using either a known item cursor or have checked that the cursor is not `CURSOR_NONE` but I don't think it'd have much impact being run for all call sites.

Reply:
It just seems odd to allow passing a `CURSOR_NONE` into this function, as it's not a valid value for an item cursor, but I'll leave it up to you if you want to refactor this further!